config.py

```diff
@@ -823,6 +823,47 @@ def _setup_auto_add_committee_to_threads(cls) -> None:
             raw_auto_add_committee_to_threads in TRUE_VALUES
         )
 
+    @classmethod
+    def _setup_auto_role(cls) -> None:
+        raw_auto_role: str = os.getenv("AUTO_ROLE", "False").lower().strip()
+
+        if not raw_auto_role:
+            cls._settings["AUTO_ROLE"] = False
+            return
+
+        if raw_auto_role not in TRUE_VALUES | FALSE_VALUES:
+            INVALID_AUTO_ROLE_MESSAGE: Final[str] = "AUTO_ROLE must be a boolean value."
+            raise ImproperlyConfiguredError(INVALID_AUTO_ROLE_MESSAGE)
+
+        cls._settings["AUTO_ROLE"] = raw_auto_role in TRUE_VALUES
+
+    @classmethod
+    def _setup_auto_roles_to_add(cls) -> None:
+        if "AUTO_ROLE" not in cls._settings:
+            INVALID_SETUP_ORDER_MESSAGE: Final[str] = (
+                "Invalid setup order: AUTO_ROLE must be set up "
+                "before AUTO_ROLES_TO_ADD can be set up."
+            )
+            raise ImproperlyConfiguredError(INVALID_SETUP_ORDER_MESSAGE)
+
+        raw_auto_roles_to_add: str = os.getenv("AUTO_ROLES_TO_ADD", default="").strip()
+
+        if not raw_auto_roles_to_add:
+            cls._settings["AUTO_ROLES_TO_ADD"] = set()
+            return
+
+        auto_roles_to_add: AbstractSet[str] = {
+            role.strip() for role in raw_auto_roles_to_add.split(",") if role.strip()
+        }
+
+        if not auto_roles_to_add:
+            INVALID_AUTO_ROLES_TO_ADD_MESSAGE: Final[str] = (
+                "AUTO_ROLES_TO_ADD must contain at least one role name."
+            )
+            raise ImproperlyConfiguredError(INVALID_AUTO_ROLES_TO_ADD_MESSAGE)
+
+        cls._settings["AUTO_ROLES_TO_ADD"] = auto_roles_to_add
+
```

Comment:
```suggestion
    @classmethod
    def _setup_auto_assign_roles(cls) -> None:
        raw_auto_assign_roles: str = os.getenv("AUTO_ASSIGN_ROLES", default="").strip()

        cls._settings["AUTO_ASSIGN_ROLES"] = {
            role_name.strip()
            for role_name in raw_auto_assign_roles.split(",")
            if role_name.strip()
        }

```
The resulting role names set can be empty, so empty string checks can be skipped as the iteration over a split empty string just results in an empty set.